third_task.py

```diff
@@ -0,0 +1,224 @@
+            if counter < rows_:
+                filtered_data.append(f_data)
+                counter += 1
+            else:
+                break
+        return filtered_data
+    else:
+        return 'need params gender or rows'
+
+
+def add_fields_to_file(file, rows):
+    """
+    :param rows: ew fields to our file
+     :param file: name of our file
+    :return rows
+    """
+    logging.info('add new fields to our file')
+    for i, row in enumerate(rows, start=1):
+        row['global_index'] = i
+        row['current_time'] = datetime.now()
+
+        match row['name.title']:
+            case 'Mrs':
+                row['name.title'] = 'missis'
+            case 'Ms':
+                row['name.title'] = 'miss'
+            case 'Mr':
+                row['name.title'] = 'mister'
+            case 'Madame':
+                row['name.title'] = 'mademoiselle'
+
+        row['dob.date'] = datetime.strptime(row['dob.date'], '%Y-%m-%dT%H:%M:%S.%fZ').strftime('%m/%d/%Y')
+        row['registered.date'] = datetime.strptime(row['registered.date'], '%Y-%m-%dT%H:%M:%S.%fZ').strftime('%m-%d-%Y/%H:%M:%S')
+
+    with open(file + '.csv', 'w', newline='', encoding='utf-8') as file:
+        csv_writer = csv.DictWriter(file)
+        csv_writer.writerows(rows)
+
+    logging.info("fields modified successfully.")
+    return rows
+
+
+def move_file_to_new_folder(destination_f):
+    """
+    :param destination_f: creating new destination folder
+
+    """
+    logging.info('creating new destination folder and move file to this folder')
+    if not os.path.exists(destination_f):
+        os.makedirs(destination_f)
+        os.replace('third_task.py', destination_f)
+        os.chdir(destination_f)
+
+
+def parse_data(data):
+    """
+
+    :param data: explored data
+    :return: structured data
+    """
+    logging.info('struct data')
+    structured_data = {}
+    for row in data:
+        dob_date = datetime.strptime(row['dob.date'], '%m/%d/%Y')
+        dob_decade = f"{dob_date.year // 10 * 10}-th"
+        country = row['location.country']
+
+        if dob_decade not in structured_data:
+            structured_data[dob_decade] = {}
+        if country not in structured_data[dob_decade]:
+            structured_data[dob_decade][country] = []
+
+        structured_data[dob_decade][country].append(row)
+
+    return structured_data
+
+
+def make_subfolders(structured_data):
+    """
+
+    :param structured_data: data structured by pattern
+    """
+    logging.info('make subfolders for each decades')
+    for decade in structured_data.keys():
+        os.makedirs(decade)
+
+
+def store_data_to_folders(structured_data):
+    """
+    :param structured_data: data structured by pattern
+
+    """
+    logging.info("Storing data to folders")
+    for decade, countries in structured_data.items():
+        for country, rows in countries.items():
+            folder = os.path.join(decade, country)
+            os.makedirs(folder)
```

Comment:
you don't need to create dir in line 148, because you'll create the whole hierarchy in line 160